### DISCUSSION TITLE
More small scenery slots in the object selection

### DISCUSSION THREAD

**Discussion by SixFrags** [OpenRCT2/OpenRCT2#20319] [Feature requests]
With an increasing amount of small scenery objects available, it would be of great help with workbench creation to have a higher limit than the 2047 limit of small scenery in the object selection. Small scenery is the most used object type by far, so I think it would make sense if this limit could be less harsh. 
I understand it requires more RAM resources used, so if it causes any instabilities it would probably not be worth it, but it would be of great help if this limit could be raised without causing any issues.

**Comment by 733737** [comment]
I think the developers (I could be wrong) want to address the cpu optimizations for now and wait a few months before changing limits. I believe that if they decide on increasing limits it will be a small increase in multiple areas, because once increased it cannot be decreased for compatibility. It may or may not also involve iterating the parkfile version. So don't expect a change in the short term, unless zehmatt finds an optimization for dynamic memory allocation. Memory isn't the main reason that the limit is 2047, however. (at least I don't think it is)

**Reply by SixFrags** [reply]
Thanks for your reply. It makes sense if they want to address cpu optimizations first. I was hoping it wouldn't be that impactful and that it was only some number changing, but reading your reply it may involve more unwanted behaviours. I still hope the devs would at least consider having more small scenery slots available. The other object types have high enough limits, so those won't really need to be raised I think.

**Reply by ZehMatt** [reply]
> I think the developers (I could be wrong) want to address the cpu optimizations for now and wait a few months before changing limits. I believe that if they decide on increasing limits it will be a small increase in multiple areas, because once increased it cannot be decreased for compatibility. It may or may not also involve iterating the parkfile version. So don't expect a change in the short term, unless zehmatt finds an optimization for dynamic memory allocation. Memory isn't the main reason that the limit is 2047, however. (at least I don't think it is)

Nothing was decided and no one said anything about optimizing the game for months.

**Reply by 733737** [reply]
that's untrue. Duncanspumpkin said no to increasing the ride limit at this time. I assumed the team has no interest in increasing limits, since they have had no discussions on it since the NSF raised the limits initially. At least no serious discussions.

**Reply by Broxzier** [reply]
Rides and scenery objects are very different things, so the opinion on one limit does not relate to another.

**Reply by 733737** [reply]
I know they are unrelated but I think an increase in one should be linked to an increase to the other for two reasons: 1.) Perhaps a higher scenery limit will inspire someone to add more themed sections and therefore more rides 2.) increasing one and not the other shows favoritism to one playstyle, ie a scenery heavy playstyle versus a millionare economy playstyle.

**Comment by ZehMatt** [comment]
What new limit do you have in mind?

**Reply by SixFrags** [reply]
I think doubling it would solve the issues with workbench creation I'm having now a lot.
Raising the limit also makes the game future-proof, because as you maybe know Tolsimir released a new Object Creator which (currently) creates small scenery .parkobj files with a very user friendly user interface and functionalities. I expect we will see a lot of new small scenery objects in the near future because of it, so raising the limit could cover that as well.

**Reply by ZehMatt** [reply]
Would need to do some basic profiling to how well all this would work, I'm not really against raising it but perhaps double is too much, will have to check what the implications are.

**Reply by SixFrags** [reply]
Yeah, doubling might be a bit too much, but whatever increase is possible without too many implications would be of great value! Thanks for even considering increasing it.

**Reply by SixFrags** [reply]
Any update on this? I'm really struggling with this cap creating a new up to date workbench. Really hope the small scenery limit be raised somewhat!

**Reply by ZehMatt** [reply]
This has currently no priority, sorry.

**Reply by SixFrags** [reply]
But do you think a number change from let's say 1000 wall slots to 1000 small scenery slots would be a quick fix for this issue? That way we would circumvent the potential memory problem, it (maybe) wouldn't cost you as much time to implement and a lot of people get a large QoL boost for the game.
I understand this is not a priority for you guys, but for a lot of people playing the game it would be such a big QoL boost.

**Reply by ZehMatt** [reply]
There is no real issue so there is no need to rush changes, this has to simply wait.

**Reply by SixFrags** [reply]
I don't mind waiting at all, was just trying to find out if you guys were interested in any way to swap some caps and if it would be hard to do. I understand it's no priority at all, and only a QoL boost, so sorry if I came over as pushing too much.

**Comment by Broxzier** [comment]
I'm surprised you hit this limit and find this not enough. This limit was already bumped up from 252 to 2047 with the new save format. That's over 8 times as much.

**Reply by SixFrags** [reply]
I may not have made it clear that I hit the limit trying to create a workbench, and that it's not for a particular park project. Workbenches are like the backbone for starting any park, with a wide variety of options available you can chose to use in the park. The main benefit of a very large workbench is that the chances of a cluttered "?" tab with tons of objects are slim (which means a lot of frustrating extra scrolling time during building) and the whole park building process is much more organized and streamlined. 

This has a small overlap with Fisch' request (https://github.com/OpenRCT2/OpenRCT2/discussions/19892), but for the short term this is a more easy solution (at least for me).

**Comment by SixFrags** [comment]
I was just thinking, that if memory usage of more small objects is a thing, wouldn't it be a good idea to shuffle some limits around (if that helps)? For example walls/large scenery/paths/land sides etc don't need that large of a cap, and instead getting a larger amount of small scenery in would be better I think.

**Reply by karst** [reply]
This would be too late now. The limit is already in place, so you cannot reduce it without the risk of breaking parks.

**Comment by karst** [comment]
I think for workbenches it might be nicer if we could just deliver parkobj files for scenery groups that have other parkobj in them.

**Reply by SixFrags** [reply]
but .park files already contain all the .dats/.parkobjs if you just load all the scenery groups into a new game and save it. Don't really see the benefit of your suggestion, other than a very large scenery group .parkobj
